File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.swift
import Foundation

func solution(_ n:Int) -> Int {
    let isEven = n % 2 == 0
    var answer = 0
    var start = isEven ? 2 : 1
    for i in stride(from: start, through: n, by: 2) {
        answer += isEven ? i * i : i
    }
    return answer
}

// simple
//func solution(_ n:Int) -> Int {
//    if n % 2 == 0 { return stride(from: 2, through: n, by: 2).reduce(0) { $0 + $1 * $1 } }
//    else { return stride(from: 1, through: n, by: 2).reduce(0, +) }
//}

print(solution(7))
print(solution(10))

File: programmers/lv0/코딩기초트레이닝/연산과조건문/홀짝에따라다른값반환하기/Solution.cs
using System;

public class Solution {
    public int solution(int n) {
        int answer = 0;
        bool isEven = n % 2 == 0;
        int start = isEven ? 2 : 1;
        for (int i = start; i <= n; i+=2)
        {
            answer += (isEven) ? i * i : i;
        }
        return answer;
    }
}